contents/elm/Elm.md

```diff
@@ -0,0 +1,256 @@
+
+Your module imports form a cycle:
+
+    ┌─────┐
+    │    Bar
+    │     ↓
+    │    Foo
+    └─────┘
+
+Learn more about why this is disallowed and how to break cycles
+here:<https://elm-lang.org/0.19.0/import-cycles>
+```
+
+- Wrong argument types:
+
+```
+> String.join 4 ["Hello", "World"]
+-- TYPE MISMATCH ---------------------------------------------------------- REPL
+
+The 1st argument to `join` is not what I expect:
+
+3|   String.join 4 ["Hello", "World"]
+                 ^
+This argument is a number of type:
+
+    number
+
+But `join` needs the 1st argument to be:
+
+    String
+
+Hint: Try using String.fromInt to convert it to a string?
+```
+
+- Syntax errors, e.g. if you try to import modules with the JavaScript syntax:
+
+```
+> import * from 'Util'
+-- EXPECTING IMPORT NAME -------------------------------------------------- REPL
+
+I was parsing an `import` until I got stuck here:
+
+2| import * from 'Util'
+          ^
+I was expecting to see a module name next, like in these examples:
+
+    import Dict
+    import Maybe
+    import Html.Attributes as A
+    import Json.Decode exposing (..)
+
+Notice that the module names all start with capital letters. That is required!
+
+Read <https://elm-lang.org/0.19.1/imports> to learn more.
+```
+
+### Benefit: Canonical Coding Style
+
+Like in Go, Elm has largely managed to avoid the style wars of older languages by having introduced a recommended style guide and code formatter (`elm-format`) early on.
+
+`elm-format` has no configuration, eliminating code style discussions with your team, and increasing the consistency of Elm code you see out in the wild.
+
+In particular, the style guide aims to reduce version control diffs. Here's an example of the recommended style for a `Record` in Elm (a `Record` is like a JavaScript object):
+
+```elm
+person =
+    { age = 18
+    , firstName = "John"
+    , lastName = "Doe"
+    }
+```
+
+Note how if you were to add a new field `favoriteLanguage` to `person`, there would only be a one line diff:
+
+```elm
+person =
+    { age = 18
+    , firstName = "John"
+    , lastName = "Doe"
+    , favoriteLanguage = "PHP"
+    }
+```
+
+If you're not used to this style (JavaScript achieves this reduced code diff via [trailing commas](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas) for example), don't worry. You can code however you're used to and then run `elm-format` to format your code automatically!
+
+That's the benefit of having one recommended style guide and using `elm-format`.
+
+Further reading:
+- [Official style guide](https://elm-lang.org/docs/style-guide)
+- [elm-format](https://github.com/avh4/elm-format)
+
+### Benefit: Package Ecosystem
+
+You can share your package, or use any of the packages available at https://package.elm-lang.org/.
+
+All packages there are written in Elm and therefore are safe (since Elm code cannot produce any side effects).
```

Comment:
can put a tool tip on safe explaining (since Elm code cannot...)

Reply:
I removed the word safe, and described it instead.